airflow/dag/serialization/json_schema.py

```diff
@@ -0,0 +1,140 @@
+                    var_schema={'type': 'number'},
+                    type_enum=[DAT.TIMEDELTA.value]),
+                'retry_exponential_backoff': {'type': 'boolean'},
+                'params': make_object_schema(
+                    var_schema={'type': 'object'},
+                    type_enum=[DAT.DICT.value]),
+                'priority_weight': {'type': 'number'},
+                'weight_rule': {'type': 'string'},
+                'executor_config': make_object_schema(
+                    var_schema={'type': 'object'},
+                    type_enum=[DAT.DICT.value]),
+                'do_xcom_push': {'type': 'boolean'},
+                # _dag field must be a dag_id.
+                '_dag': make_object_schema(
+                    var_schema={'type': 'string'},
+                    type_enum=[DAT.DAG.value]),
+                'ui_color': {'type': 'string'},
+                'ui_fgcolor': {'type': 'string'},
+                'template_fields': {
+                    'anyOf': [
+                        {
+                            'type': 'array',
+                            'items': {'type': 'string'}
+                        },
+                        make_object_schema(
+                            var_schema={'type': 'array'},
+                            type_enum=[DAT.TUPLE.value]),
+                    ]
+                }
+            },
+            'required': [
+                'task_id', 'owner', 'start_date', '_dag',
+                'ui_color', 'ui_fgcolor', 'template_fields']
+        },
+        type_enum=[DAT.OP.value])
+
+
+def make_dag_schema() -> dict:
+    """jsonschema of a serialized DAG."""
+    return make_object_schema(
+        var_schema={
+            'type': 'object',
+            'properties': {
+                'default_args': make_object_schema(var_schema={'type': 'object'}),
+                'params': make_object_schema(var_schema={'type': 'object'}),
+                '_dag_id': {'type': 'string'},
+                'task_dict': make_object_schema(
+                    var_schema={
+                        'type': 'object',
+                        'additionalProperties': make_operator_schema()
+                    },
+                    type_enum=[DAT.DICT.value]),
+                'timezone': make_object_schema(
+                    var_schema={'type': 'string'},
+                    type_enum=[DAT.TIMEZONE.value]),
+                'schedule_interval': {
+                    'anyOf': [
+                        {'type': 'string'},  # cron format.
+                        make_object_schema(
+                            var_schema={'type': 'number'},
+                            type_enum=[DAT.TIMEDELTA.value])]
+                },
+                'catchup': {'type': 'boolean'},
+                'is_subdag': {'type': 'boolean'}
+            },
+            'required': [
+                'default_args', 'params',
+                '_dag_id', 'fileloc', 'task_dict']
```

Comment:
`'additional_properties': False` too I think?

Reply:
updated in https://github.com/apache/airflow/pull/5743